src/Pagination/PaginateDirective.php

```diff
@@ -110,10 +121,24 @@ public function manipulateFieldDefinition(DocumentAST &$documentAST, FieldDefini
     public function resolveField(FieldValue $fieldValue): FieldValue
     {
         $fieldValue->setResolver(function ($root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo): Paginator {
+            if ($this->directiveHasArgument('resolver')) {
+                $builderResolver = $this->getResolverFromArgument('resolver');
+
+                $query = $builderResolver($root, $args, $context, $resolveInfo);
+
+                assert(
+                    $query instanceof LengthAwarePaginator,
```

Comment:
Actually, the return value of this method only says `Illuminate\Contracts\Pagination\Paginator` - I think it would not work with that though, given `PaginatorField` and `ConnectionField` expect a `LengthAwarePaginator`. So I think we should probably change the return value?

Reply:
I went through this now seems like `Illuminate\Pagination\LengthAwarePaginator` implements `Illuminate\Contracts\Pagination\LengthAwarePaginator` which extends `Illuminate\Contracts\Pagination\Paginator` so `Illuminate\Pagination\LengthAwarePaginator` is actually of typeof `Illuminate\Contracts\Pagination\Paginator` so I guess it made more sense if I had written:

```php
                assert(
                    $query instanceof Paginator,
                    "The method referenced by the resolver argument of the @{$this->name()} directive on {$this->nodeName()} must return a Paginator."
                );
```

instead, that means the return type doesn't need to be changed tried with `Illuminate\Pagination\Paginator` instead of `Illuminate\Pagination\LengthAwarePaginator` which also worked.

Reply:
For now I committed it the way I described above if you see anything wrong with it please let me know.

Reply:
We should define the type to be as wide as possible, but as narrow as needed. The downstream resolvers require `Illuminate\Contracts\Pagination\LengthAwarePaginator`, so that is what we should expect.

Reply:
> We should define the type to be as wide as possible, but as narrow as needed. The downstream resolvers require `Illuminate\Contracts\Pagination\LengthAwarePaginator`, so that is what we should expect.

Wasn't aware that, that was used internally so does this look ok?

```php
        $fieldValue->setResolver(function ($root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo): LengthAwarePaginator {
            if ($this->directiveHasArgument('resolver')) {
                $paginator = $this->getResolverFromArgument('resolver')($root, $args, $context, $resolveInfo);

                assert(
                    $paginator instanceof LengthAwarePaginator,
                    "The method referenced by the resolver argument of the @{$this->name()} directive on {$this->nodeName()} must return a LengthAwarePaginator."
                );

                return $paginator;
            }
```